src/features/auth/login.ts

```diff
@@ -23,6 +24,11 @@ export const login =
 				if (!user) {
 					throw new NotFoundError('User not found.');
 				}
+				const userPermissions = await permissions.getUserPermissions(user.id);
```

Comment:
I would put this after the password check, that way if that fails we avoid a db request. Although also this check could potentially be in the `onLogin` if it's unused in open-balena?

Reply:
@Page- It is not directly used on open-balena-api right now but I could see it being used in the future, more over, credentials_login is something of the default-user that exists already on open-balena so in my mind it makes sense to be here instead of on the api callback